### PR TITLE
Finessing configurator submission code

### PR DESCRIPTION
**Pull request by dpaks**
Lb @pkharat and VPN @SureshDH  yet to be merged to this branch.

### CODE REVIEW

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py [hunk 1]
```diff
@@ -192,51 +183,56 @@ def _test_handle_event(self, ev):
...
-                if periodicity == gen_cfg_const.INITIAL_HM_RETRIES:
+            elif const.EVENT_CONFIGURE_HEALTHMONITOR in ev.id:
+                if periodicity == const.INITIAL_HM_RETRIES:
                     mock_hm_poll_event.assert_called_with(
-                                ev, max_times=gen_cfg_const.INITIAL_HM_RETRIES)
-                elif periodicity == gen_cfg_const.FOREVER:
+                                ev, max_times=const.INITIAL_HM_RETRIES)
+                elif periodicity == const.FOREVER:
                     mock_hm_poll_event.assert_called_with(ev)
-            elif ev.id == 'CLEAR_HEALTHMONITOR':
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:238]: Why is clear healthmonitor completely removed?

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py [hunk 2]
```diff
@@ -331,22 +326,10 @@ def test_configure_hm_forever_genericconfigeventhandler(self):
...
         ev = fo.FakeEventGenericConfig()
-        ev.data['resource_data'].update({'periodicity': gen_cfg_const.FOREVER})
+        ev.data['resource_data'].update({'periodicity': const.FOREVER})
         ev.id = 'CONFIGURE_HEALTHMONITOR forever'
         self._test_handle_event(ev)
 
-    def test_clear_hm_genericconfigeventhandler(self):
-        """ Implements test case for clear health monitor method
-        of generic config event handler.
-
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:341]: Why is this removed?

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py [hunk 3]
```diff
@@ -356,9 +339,5 @@ def test_handle_configure_healthmonitor_genericconfigeventhandler(self):
         """
 
         ev = fo.FakeEventGenericConfig()
-        ev.id = 'CONFIGURE_HEALTHMONITOR'
+        ev.id = const.EVENT_CONFIGURE_HEALTHMONITOR
         self._test_handle_periodic_event(ev)
-
-
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:362]: Is this not needed because we moved to neutron base unit test?

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+        Returns: none
+
+        """
+
+        agent, sc = self._get_NfpServiceRpcManager_object()
+        arg_dict = {'context': self.fo.context,
+                    'resource_data': self.fo.kwargs}
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py]: Lets get rid of "\" here.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        super(NfpServiceEventHandlerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = fo.FakeObjects()
+        self.context = {'notification_data': {},
+                        'resource': 'interfaces'}
+
+    @mock.patch(__name__ + '.fo.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.fo.FakeObjects.drivers')
+    @mock.patch(__name__ + '.fo.FakeObjects.sc')
+    def _get_NfpServiceEventHandler_object(self, sc, drivers, rpcmgr):
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py]: Can we name this function without uppercase chars. Follow the same convention as other function names.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        """
+
+        agent, sc = self._get_NfpServiceEventHandler_object()
+        driver = mock.Mock()
+
+        with mock.patch.object(
+                driver, 'run_heat', return_value=result) as mock_config_inte, (
+             mock.patch.object(
+                agent, '_get_driver', return_value=driver)):
+
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py:126]: Avoid too many spaces like this in the code. Not sure if PEP will complain, but keep one line gap which should be enough.

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/firewall/test_vyos_fw_driver.py [hunk 7]
```diff
@@ -40,25 +37,75 @@ def __init__(self, *args, **kwargs):
...
+            self.driver.configure_interfaces(self.fo.context, self.kwargs)
+
+            data = jsonutils.dumps(self.fo.static_ip_data())
+            mock_post.assert_called_with(self.fo.url_for_add_static_ip,
+                                         data=data,
+                                         timeout=self.fo.timeout)
+
     def test_configure_interfaces(self):
-        """ Implements test case for configure interfaces method
+        """ Implements test case for configure log forward method
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/firewall/test_vyos_fw_driver.py]: Correct the comment. It is configuration of interfaces.

File: gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py [hunk 8]
```diff
@@ -265,8 +285,8 @@ def _fake_resource_data(self):
                     'stitching_cidr': '192.168.0.0/28',
                     'destination_cidr': '192.168.0.0/28',
                     'stitching_mac': '00:0a:95:9d:68:16',
-                    'provider_interface_index': 'provider_interface_index',
-                    'stitching_interface_index': 'stitching_interface_index',
+                    'provider_interface_index': '2',
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:288]: Are these values taken just for example? We removed the hardcoded valued form orchestrator right?

File: gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py [hunk 9]
```diff
@@ -344,3 +364,8 @@ def __init__(self):
                     'host': fo.host,
                     'resource_data': kwargs}
         self.id = 'dummy'
+
+
+class FakeEventGetNotifications(object):
```
Comment (njagadish) [inline gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:369]: Lets add comment for this class

File: gbpservice/nfp/configurator/Dockerfile [hunk 10]
```diff
@@ -47,21 +47,14 @@ RUN git clone -b stable/mitaka --single-branch https://github.com/openstack/neut
...
-
+RUN git clone https://github.com/openstack/group-based-policy.git group-based-policy
+RUN cd /group-based-policy && git fetch https://git.openstack.org/openstack/group-based-policy GIT-BRANCH-NAME && git checkout FETCH_HEAD
 RUN cp -r /group-based-policy/gbpservice /usr/local/lib/python2.7/dist-packages/
 RUN cp /group-based-policy/gbpservice/nfp/bin/nfp /usr/bin/
 RUN chmod +x /usr/bin/nfp
 RUN cp /group-based-policy/gbpservice/nfp/bin/nfp_configurator.ini /etc/
 RUN sed -i "s/log_forward_ip_address=*.*/log_forward_ip_address=VIS_VM_IP_ADDRESS/" /etc/nfp_configurator.ini
 RUN cp /group-based-policy/gbpservice/nfp/bin/policy.json /etc/
-RUN cp -r /group-based-policy/gbpservice/nfp/configurator/config /etc/nfp_config
```
Comment (njagadish) [inline gbpservice/nfp/configurator/Dockerfile:64]: Why is this not needed now?

File: gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py [hunk 11]
```diff
@@ -565,43 +500,28 @@ def create_firewall(self, context, firewall, host):
...
+            LOG.error(err_msg)
             return common_const.STATUS_ERROR
 
+        if resp is common_const.STATUS_SUCCESS:
+            msg = ("Configured firewall successfully for service at %r." % url)
+            LOG.info(msg)
+            return common_const.STATUS_ACTIVE
+
+        err_msg += (("Reason: %r, Response Content: %r" %
+                     (resp.pop('message'), resp))
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py:519]: If there is a possibility of resp becoming None, there will be an exception. Please check the possibility.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py (hunk 3 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:362]: This is unnecessary for run_tests script. We do not use unittest but basetest from neutron tests.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py (hunk 4 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py]: ### Addressed.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py (hunk 5 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py]: ### Addressed.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py (hunk 6 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_nfp_service.py:126]: Jagadish, there is only one line gap.

File: gbpservice/neutron/tests/unit/nfp/configurator/drivers/firewall/test_vyos_fw_driver.py (hunk 7 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/drivers/firewall/test_vyos_fw_driver.py]: ### Addressed.

File: gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py (hunk 8 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:288]: We never removed it from configurator side. We have to remove these from driver as well as schema files.

File: gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py (hunk 9 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/test_data/fw_test_data.py:369]: ### Addressed.

File: gbpservice/nfp/configurator/Dockerfile (hunk 10 above)
Comment (dpaks) [inline gbpservice/nfp/configurator/Dockerfile:64]: This is not needed because this dir is used only by ASAv which we are not going to submit. Hence, we though of removing this empty directory for submission.

File: gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py (hunk 11 above)
Comment (dpaks) [inline gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py:519]: Resp is constructed in the RestApi class. It can either be a string or a dict. I don't think that this will be an issue. Checked it.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py (hunk 1 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:238]: This was removed as it was just a placeholder. Anyways, I have written a test case for this.

File: gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py (hunk 2 above)
Comment (dpaks) [inline gbpservice/neutron/tests/unit/nfp/configurator/agents/test_generic_config.py:341]: This was removed as it was just a placeholder with the child function having no definition. Anyways, I have written a test case for this.
